alchemiscale/interface/client.py

```diff
@@ -119,16 +122,25 @@ def create_network(
 
         validate_network_nonself(network)
 
-        from rich.progress import Progress
-
         sk = self.get_scoped_key(network, scope)
-        with Progress(*self._rich_waiting_columns(), transient=False) as progress:
-            task = progress.add_task(f"Submitting [bold]'{sk}'[/bold]...", total=None)
 
+        def post():
             data = dict(network=network.to_dict(), scope=scope.dict())
-            scoped_key = self._post_resource("/networks", data, compress=compress)
-            progress.start_task(task)
-            progress.update(task, total=1, completed=1)
+            return self._post_resource("/networks", data, compress=compress)
+
+        if visualize:
+            from rich.progress import Progress
+
+            with Progress(*self._rich_waiting_columns(), transient=False) as progress:
+                task = progress.add_task(
+                    f"Submitting [bold]'{sk}'[/bold]...", total=None
+                )
+
+                scoped_key = post()
+                progress.start_task(task)
+                progress.update(task, total=1, completed=1)
```

Comment:
For reference, the reason we do `start_task` *after* we actually perform the operation is that we really just care about showing how much time has elapsed to the user as part of our indication that the client is busy doing something. We then immediately set progress to complete (`progress.update`) after "starting" it (`progress.start`) as outlined here: https://rich.readthedocs.io/en/stable/progress.html#indeterminate-progress